### ISSUE TITLE
[HOLD for payment 2025-01-22] [$250] Update react-native-qrcode-svg to not use text-decoding

### ISSUE THREAD

**Issue by gedu** [Expensify/App#54096]
### Background
react-native-qrcode-svg relies on the text-encoding library to provide TextDecoder and TextEncoder functionality, which is essential for encoding and decoding text data. However, starting with React Native 0.75, these functionalities are natively supported, making the inclusion of text-encoding redundant for modern environments.
Despite this, the library currently includes text-encoding in all builds, regardless of the React Native version. This results in an unnecessary increase in app bundle size, particularly for developers using React Native 0.75 or higher, where native support for TextDecoder and TextEncoder is already available. This behavior impacts performance and resource usage, especially for applications targeting modern devices.
Reference the relevant issue and commit:
• Issue: Expensify/react-native-qrcode-svg#199
• Commit: 53f1c5f

### Problem
The react-native-qrcode-svg library always includes the text-encoding library, adding approximately 600KB to the app bundle size. This is unnecessary for React Native 0.75+ environments, where TextDecoder and TextEncoder are natively supported, leading to inefficiencies in modern applications.

### Solution
Update react-native-qrcode-svg to conditionally include the text-encoding library based on the React Native version or the availability of native TextDecoder and TextEncoder. This can be achieved through the following steps:
Version Check:
○ Add conditional logic to detect if the app is running on React Native 0.75 or higher.
○ Exclude the text-encoding library from the bundle if native support for TextDecoder and TextEncoder is available.
Metro Serializer Optimization:
○ Use tools like Metro Serializer Esbuild to further optimize the bundle and ensure unused dependencies like text-encoding are excluded.

Slack: https://expensify.slack.com/archives/C05LX9D6E07/p1734018386310029

Note: Given that we are on 0.75, we can remove text-encoding lib
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021869155133024534590</li>
        <li>Upwork Job ID: 1869155133024534590</li>
        <li>Last Price Increase: 2024-12-17</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @isabelastisser</details>

**Comment by gedu** [comment]
Tasks
- [ ] Remove text-decoding from react-native-qrcode-svg 
- [ ] Update Exfy app to use the new react-native-qrcode-svg version

**Comment by gedu** [comment]
I confirmed that removing the line:
```
global.TextEncoder = require("text-encoding").TextEncoder;
```
our app will tree shake that lib and won't be on the final bundle. Now I'm looking into a way to adding that line for old RN versions

**Comment by AbdulrhmnGhanem** [comment]
Hi @gedu, I see you are working on the issue already, I saw it yesterday and wanted to experiment with it before commenting. Feel free to continue on with my PR!

**Comment by gedu** [comment]
@AbdulrhmnGhanem Hey, thanks for your PR. The main idea of the issue is to remove the text-encoder to reduce the bundle size, your approach always keeps the lib into the bundle. So it won't work.

**Comment by AbdulrhmnGhanem** [comment]
It doesn't get bundled unless the library users install it. I am curious
though if there's a solution that doesn't require modifying metro config
for each app.

**Comment by gedu** [comment]
@mountiny  I created a PR using the metro solution, on Exfy app the only thing we have to do is increase the version and that's all.

**Comment by isabelastisser** [comment]
### FYI, I will be OOO from Dec 23 to Jan 6, so please reassign the issue to another team member for urgency, IF needed.

**Comment by mountiny** [comment]
@gedu  can you link the pr here please?

**Comment by gedu** [comment]
Yep: https://github.com/Expensify/react-native-qrcode-svg/pull/227

**Comment by mountiny** [comment]
thank you! @thesahindia are you able to review it and test it?

**Comment by gedu** [comment]
@mountiny Updated the Example app RN version: https://github.com/Expensify/react-native-qrcode-svg/pull/228

**Comment by mountiny** [comment]
Reviewed

**Comment by isabelastisser** [comment]
@gedu @parasharrajat, can you please provide an update? Thanks!

**Comment by isabelastisser** [comment]
Bump @gedu @parasharrajat for an update. Thanks!

**Comment by gedu** [comment]
I'm looking into a possible Example App issue, and will continue updating the version

**Comment by mountiny** [comment]
Fixing the publishing flow that seems broken

**Comment by gedu** [comment]
PR created: https://github.com/Expensify/App/pull/55080

**Comment by mountiny** [comment]
@gedu anything else left to do here?

**Comment by mountiny** [comment]
So only payments to @parasharrajat

**Comment by mountiny** [comment]
I think we should already have tests like this

**Comment by isabelastisser** [comment]
Payment summary:

@parasharrajat requires payment through NewDot Manual Requests $250 for C+ review

**Comment by JmillsExpensify** [comment]
$250 approved for @parasharrajat
